zstd_vendor/CMakeLists.txt

```diff
@@ -39,7 +39,9 @@ macro(build_zstd)
       -DZSTD_BUILD_STATIC=OFF
       -DZSTD_BUILD_SHARED=ON
       -DZSTD_BUILD_PROGRAMS=OFF
-      ${extra_cmake_args})
+      ${extra_cmake_args}
+    # Note: zstd v1.4.6 will include the following fix. When that is released, upgrade and remove this patch.
+    PATCH_COMMAND patch -p1 -d . < ${CMAKE_CURRENT_SOURCE_DIR}/cmake_minimum_required_2.8.12.patch)
```

Comment:
should something like in https://github.com/ros2/rviz/blob/2206b9257716c05dc4d5eaa7ffae12e434f151ca/rviz_ogre_vendor/CMakeLists.txt#L157 be done here?

when setting up a windows machine for debugging an issue, I needed to install patch from choco but the places that were using `find_package(Patch)` were working

Reply:
https://github.com/ros2/rosbag2/issues/632 is the planned solution, which should also work well without this dependency

Reply:
Sounds good, we will need something like that before the Galactic release as the current build from source instructions doesn't seem to be working on Windows